### PR TITLE
[Concept Entry] Sklearn Biclustering

### PR DESCRIPTION
**Pull request by SaviDahegaonkar**
### Description

Created a new entry on the Biclustering concept under Sklearn.

### Issue Solved
#5202 

### Type of Change

- Adding a new entry

### Checklist

- [ ] All writings are my own.
- [ ] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [ ] I have performed a self-review of my own writing and code.
- [ ] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/sklearn/concepts/biclustering/biclustering.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+Title: 'Biclustering'
+Description: 'A technique for grouping rows and columns of a matrix to discover local patterns in data.'
+Subjects:
+  - 'Data Science'
+  - 'Data Visualization'
+  - 'Machine Learning'
+Tags:
+  - 'Clustering'
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/biclustering/biclustering.md:9]: 'Clustering' is not a valid tag. Refer to [tags.md](https://github.com/Codecademy/docs/blob/main/documentation/tags.md) for reference.

```suggestion
```

File: content/sklearn/concepts/biclustering/biclustering.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+# For Spectral Bi-clustering
+model = SpectralBiclustering(n_clusters=number_of_biclusters, method="log", random_state=seed)
+model.fit(data_matrix)
+```
+
+- `n_clusters`: Number of biclusters to create.
+- `random_state`: Ensures the randomness for reproducible results.
+- `method:`(For SpectralBiclustering) Specifies the algorithm variant, e.g., "log" or "bistochastic."
+
+> **Note**: Since Bicluster is not directly available in sklearn directly, alternative methods for biclustering, such as SpectralBiclustering can be used.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/biclustering/biclustering.md:40]: There is repetition in this sentence:

```suggestion
> **Note**: Since Bicluster is not directly available in sklearn, alternative methods for biclustering, such as `SpectralBiclustering`, can be used.
```

File: content/sklearn/concepts/biclustering/biclustering.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+## Example
+
+Here's an example of biclustering using `SpectralBiclustering` from sklearn:
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/biclustering/biclustering.md:44]:
```suggestion
Here's an example of implementing biclustering using `SpectralBiclustering` from sklearn:
```

File: content/sklearn/concepts/biclustering/biclustering.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+row_labels = model.rows_
+column_labels = model.columns_
+
+# Print biclusters
+print("Row Biclusters:", row_labels)
+print("Column Biclusters:", column_labels)
+
+```
+
+The code results in the following output:
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/biclustering/biclustering.md:70]:
```suggestion
The above code results in the following output:
```

File: content/sklearn/concepts/biclustering/biclustering.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+```shell
+Row Biclusters: [[ True  True False False]
+ [ True  True False False]
+ [False False  True  True]
+ [False False  True  True]]
+Column Biclusters: [[ True  True False False]
+ [False False  True  True]
+ [ True  True False False]
+ [False False  True  True]]
+
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/biclustering/biclustering.md:81]: The output I got when running the above code has different values. Please check the code or update the output. 

Here's the output I got:

```suggestion
Row Biclusters: [[False False  True  True]
 [False False  True  True]
 [ True  True False False]
 [ True  True False False]]
Column Biclusters: [[False False  True  True]
 [ True  True False False]
 [False False  True  True]
 [ True  True False False]]
```

File: content/sklearn/concepts/biclustering/biclustering.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+model.fit(data_matrix)
+
+# For Spectral Bi-clustering
+model = SpectralBiclustering(n_clusters=number_of_biclusters, method="log", random_state=seed)
+model.fit(data_matrix)
+```
+
+- `n_clusters`: Number of biclusters to create.
+- `random_state`: Ensures the randomness for reproducible results.
+- `method:`(For SpectralBiclustering) Specifies the algorithm variant, e.g., "log" or "bistochastic."
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/biclustering/biclustering.md:38]:
```suggestion
- `method`(For SpectralBiclustering): Specifies the algorithm variant, e.g., `log` or `bistochastic`.
```

File: content/sklearn/concepts/biclustering/biclustering.md (hunk 5 above)
Comment (PragatiVerma18) [reply content/sklearn/concepts/biclustering/biclustering.md:81]: Add a brief comment in the code explaining what the binary arrays represent (e.g., `True` indicates membership in the bicluster).


```
The `rows_` and `columns_` attributes in `SpectralBiclustering` output binary arrays (or boolean masks). These indicate membership in a bicluster for each row and column. For example:

- In the **Row Biclusters**, `True` in a position means that the corresponding row is part of the bicluster.
- Similarly, in the **Column Biclusters**, `True` indicates that the corresponding column is part of the bicluster.
```

File: content/sklearn/concepts/biclustering/biclustering.md (hunk 6 above)
Comment (PragatiVerma18) [reply content/sklearn/concepts/biclustering/biclustering.md:38]: It would be helpful to briefly explain the different methods ("bistochastic", "log", etc.) and when each one is appropriate.

```
The `log` method applies logarithmic scaling, while `bistochastic` normalizes rows and columns. The choice of method can affect the results depending on the dataset."
```

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]
Hey @SaviDahegaonkar, thanks for adding this entry. I have some suggestions, I have added in comments. Please check and do the required changes. Let me know if you need any help. Happy contributing! 😄

**Review by PragatiVerma18 — APPROVED** [review]
